suc/pkg/host/common.go

```diff
@@ -0,0 +1,151 @@
+package host
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultWinsPath                    = "c:\\Windows\\wins.exe"
+	defaultWinsUsrLocalBinPath         = "c:\\usr\\local\\bin\\wins.exe"
+	defaultConfigDir                   = "c:\\etc\\rancher\\wins"
+	fileOperationAttempts              = 5
+	fileOperationAttemptDelayInSeconds = 5
+
+	// skipBinaryUpgradeEnvVar prevents the suc image from attempting to upgrade the wins binary.
+	// This is primarily used in CI, to allow for test cases to run without having to completely
+	// install rancher-wins.
+	skipBinaryUpgradeEnvVar = "CATTLE_WINS_SKIP_BINARY_UPGRADE"
+)
+
+// getRancherWinsVersionFromBinary executes the wins.exe binary located at 'path' and passes the '--version'
+// flag. The release version or commit hash is returned. If the binary returns unexpected output,
+// was built with a dirty commit, or does not exist, an error will be returned.
+func getRancherWinsVersionFromBinary(path string) (string, error) {
+	if path == "" {
+		return "", fmt.Errorf("must specify a path")
```

Comment:
IMO I prefer declaring errors as private package variables with `errors.New` so it's easy to write unit tests for them with `errors.Is`. You have some tests for this function that just expect an error being returned, and I know that realistically there isn't much of a use to do so, but personally I like to check the exact error that is being returned if at all possible. I won't block on this, but I've found it's generally a good practice, although it can conflict with writing table style tests.

Reply:
I agree with the sentiment, however this particular function is tested via the pester integration tests, so we wouldn't be able to leverage new error specific variables. Additionally a lot of the errors include runtime specific information (like paths and whatnot) that we don't know at compile time.